CMakeLists.txt

```diff
@@ -563,6 +563,12 @@ if(${CMAKE_INSTALL_PREFIX} STREQUAL /usr/local)
   set(CMAKE_INSTALL_PREFIX /usr)
 endif()
 
+option(WITH_TESTS "build with tests" ON)
+if(NOT WITH_TESTS)
```

Comment:
If someone adds code to the end of this file that is not related to tests, the return over here will break things in hard to debug ways.

I'd rather keep the testing code surrounded by WITH_TESTS instead of relying on the ordering.